x/gamm/client/cli/tx.go

```diff
@@ -384,6 +385,50 @@ func NewBuildCreatePoolMsg(clientCtx client.Context, txf tx.Factory, fs *flag.Fl
 		FuturePoolGovernor: pool.FutureGovernor,
 	}
 
+	if (pool.SmoothWeightChangeParams != smoothWeightChangeParamsInputs{}) {
+		duration, err := time.ParseDuration(pool.SmoothWeightChangeParams.Duration)
+		if err != nil {
+			return txf, nil, fmt.Errorf("could not parse duration: %w", err)
+		}
+
+		targetPoolAssetCoins, err := sdk.ParseDecCoins(pool.SmoothWeightChangeParams.TargetPoolWeights)
+		if err != nil {
+			return txf, nil, err
+		}
+
+		var targetPoolAssets []types.PoolAsset
+		for i := 0; i < len(targetPoolAssetCoins); i++ {
+
+			if targetPoolAssetCoins[i].Denom != poolAssetCoins[i].Denom {
+				return txf, nil, errors.New("initial pool weights and target pool weights should have same denom order")
```

Comment:
This isn't actually enforced by the state machine, the state machine will handle sorting. Probably more convenient to also enforce here though